pytorch_lightning/accelerators/gpu.py

```diff
@@ -53,6 +64,58 @@ def set_nvidia_flags(local_rank: int) -> None:
         devices = os.getenv("CUDA_VISIBLE_DEVICES", all_gpu_ids)
         _log.info(f"LOCAL_RANK: {local_rank} - CUDA_VISIBLE_DEVICES: [{devices}]")
 
+    def get_device_stats(self, device: Optional[torch.device] = None) -> Dict[str, Any]:
+        """Gets stats for the given GPU device."""
+        if _TORCH_GREATER_EQUAL_1_8:
+            return torch.cuda.memory_stats(device=device)
+        else:
+            gpu_stat_keys = [
+                ("utilization.gpu", "%"),
+                ("memory.used", "MB"),
+                ("memory.free", "MB"),
+                ("utilization.memory", "%"),
+                ("fan.speed", "%"),
+                ("temperature.gpu", "°C"),
+                ("temperature.memory", "°C"),
+            ]
+            gpu_stats = self._get_gpu_stats([k for k, _ in gpu_stat_keys])
+            device_stats = self._parse_gpu_stats(self._device_ids, gpu_stats, gpu_stat_keys)
+            return device_stats
```

Comment:
this doesn't use the `device` passed in anywhere?

Reply:
it does in line 70: `return torch.cuda.memory_stats(device=device)`

Reply:
but afterward, even if someone isn't using PyTorch 1.8+, they'd expect the `device` argument to still be respected and for the method to get stats only for that particular device

Reply:
good point, I was a bit confused. I updated the PR so that it does that, and tested with https://github.com/daniellepintz/pytorch-lightning/pull/2